src/editor/Renderer.cpp

```diff
@@ -1557,7 +1657,104 @@ void Renderer::setupView() {
 	view.translate(-cameraOrigin.x, -cameraOrigin.z, cameraOrigin.y);
 }
 
+void Renderer::reloadBspModels()
+{
+	isModelsReloading = true;
+
+	if (!mapRenderers.size())
+	{
+		isModelsReloading = false;
+		return;
+	}
+
+	int modelcount = 0;
+
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		if (mapRenderers[i]->map->is_model)
+		{
+			modelcount++;
+		}
+	}
+
+	if (modelcount == mapRenderers.size())
+	{
+		isModelsReloading = false;
+		return;
+	}
+
+	std::vector<BspRenderer*> sorted_renders;
+
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		if (!mapRenderers[i]->map->is_model)
+		{
+			sorted_renders.push_back(mapRenderers[i]);
+		}
+		else
+		{
+			delete mapRenderers[i];
+		}
+	}
+
+	mapRenderers = sorted_renders;
+
+	vector<string> tryPaths = {
+		"./"
+	};
+
+	tryPaths.insert(tryPaths.end(), g_settings.resPaths.begin(), g_settings.resPaths.end());
+
+	for (auto const& bsprend : mapRenderers)
+	{
+		for (auto const& entity : bsprend->map->ents)
+		{
+			if (entity->hasKey("model"))
+			{
+				std::string modelPath = entity->keyvalues["model"];
+				if (modelPath.find(".bsp") != std::string::npos)
+				{
+					for (int i = 0; i < tryPaths.size(); i++) {
+						string tryPath = tryPaths[i] + modelPath;
+						string tryPath_full = g_settings.gamedir + tryPaths[i] + modelPath;
+						if (fileExists(tryPath)) {
+							Bsp* tmpBsp = new Bsp(tryPath);
+							tmpBsp->is_model = true;
+							if (tmpBsp->valid)
+							{
+								BspRenderer* mapRenderer = new BspRenderer(tmpBsp, bspShader, fullBrightBspShader, colorShader, pointEntRenderer);
+								mapRenderers.push_back(mapRenderer);
+							}
+							break;
+						}
+						if (fileExists(tryPath_full)) {
+							Bsp* tmpBsp = new Bsp(tryPath_full);
+							tmpBsp->is_model = true;
+							if (tmpBsp->valid)
+							{
+								BspRenderer* mapRenderer = new BspRenderer(tmpBsp, bspShader, fullBrightBspShader, colorShader, pointEntRenderer);
+								mapRenderers.push_back(mapRenderer);
+							}
+							break;
+						}
+					}
+				}
+			}
+		}
+	}
+
+	isModelsReloading = false;
+}
+
 void Renderer::addMap(Bsp* map) {
+	if (!map->is_model)
+	{
+		for (int i = 0; i < mapRenderers.size(); i++)
+		{
+			mapRenderers[i]->map->is_selected = false;
+		}
+		map->is_selected = true;
+	}

```

Comment:
just noticing this `is_selected` code. I don't think that needs to exist. The `pickInfo` var indicates which map is selected.

Reply:
I replaced pickinfo because it not easy to access maps